This file is part of MXE. See LICENSE.md for licensing information.

Add host_hooks support for aarch64-apple-darwin.

When building GCC as a cross-compiler on Apple Silicon (aarch64-darwin),
the host_hooks symbol is not defined, causing linker errors. This patch
adds the necessary host configuration for aarch64-darwin hosts.

See: https://github.com/mxe/mxe/issues/2961

From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: MXE Contributors
Date: Wed, 29 Jan 2025 00:00:00 +0000
Subject: [PATCH] Add host_hooks support for aarch64-apple-darwin

diff --git a/gcc/config.host b/gcc/config.host
index 1111111..2222222 100644
--- a/gcc/config.host
+++ b/gcc/config.host
@@ -96,6 +96,13 @@ case ${host} in
     ;;
 esac
 
+case ${host} in
+  aarch64*-*-darwin* | arm64*-*-darwin*)
+    out_host_hook_obj="${out_host_hook_obj} host-aarch64-darwin.o"
+    host_xmake_file="${host_xmake_file} aarch64/t-aarch64-darwin"
+    ;;
+esac
+
 case ${host} in
   aarch64*-*-freebsd* | aarch64*-*-linux* | aarch64*-*-fuchsia*)
     case ${target} in
diff --git a/gcc/config/aarch64/host-aarch64-darwin.c b/gcc/config/aarch64/host-aarch64-darwin.c
new file mode 100644
index 0000000..1111111
--- /dev/null
+++ b/gcc/config/aarch64/host-aarch64-darwin.c
@@ -0,0 +1,28 @@
+/* aarch64-darwin host-specific hook definitions.
+   Copyright (C) 2021-2025 Free Software Foundation, Inc.
+
+This file is part of GCC.
+
+GCC is free software; you can redistribute it and/or modify it
+under the terms of the GNU General Public License as published by the
+Free Software Foundation; either version 3, or (at your option) any
+later version.
+
+GCC is distributed in the hope that it will be useful, but WITHOUT
+ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+for more details.
+
+You should have received a copy of the GNU General Public License
+along with GCC; see the file COPYING3.  If not see
+<http://www.gnu.org/licenses/>.  */
+
+#define IN_TARGET_CODE 1
+
+#include "config.h"
+#include "system.h"
+#include "coretypes.h"
+#include "hosthooks.h"
+#include "hosthooks-def.h"
+
+const struct host_hooks host_hooks = HOST_HOOKS_INITIALIZER;
diff --git a/gcc/config/aarch64/t-aarch64-darwin b/gcc/config/aarch64/t-aarch64-darwin
new file mode 100644
index 0000000..2222222
--- /dev/null
+++ b/gcc/config/aarch64/t-aarch64-darwin
@@ -0,0 +1,3 @@
+# Support file for aarch64-darwin host
+host-aarch64-darwin.o : $(srcdir)/config/aarch64/host-aarch64-darwin.c
+	$(COMPILE) $<
